azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -66,6 +76,8 @@ def __init__(self,
         self._function_context: FunctionContext = FunctionContext(**kwargs)
         self._sequence_number = 0
         self._replay_schema = ReplaySchema(upperSchemaVersion)
+        if upperSchemaVersionNew is not None and upperSchemaVersionNew > self._replay_schema.value:
+            self._replay_schema = ReplaySchema(upperSchemaVersionNew)
```

Comment:
If you have `ReplaySchema(999)`, won't this break as `ReplaySchema` only supports 1 through 3?
Might be good adding some kind of guard if so.

Reply:
Good point

Reply:
In these cases, the expected behavior is for the SDK to use it's highest supported schema version, as the parsing failing implies the extension is "too far ahead" of the SDK, and the extension supports all prior ooproc protocol versions.

Reply:
Added a check using ReplaySchema._value2member_map_. This was the most elegant way to do it, but Enum._value2member_map_ is undocumented. Happy to change it to something like [enum_member.value for enum_member in ReplaySchema]

Reply:
Actually updated to use this approach as it made getting the max version easier anyway